test/unit/gateways/deepstack_test.rb

```diff
@@ -0,0 +1,223 @@
+    @gateway.expects(:ssl_post).returns(successful_purchase_response)
+
+    response = @gateway.purchase(@amount, @credit_card, @options)
+    assert_success response
+
+    assert_equal 'ch_IoSx345fOU6SP67MRXgqWw', response.authorization
+    # assert response.test?
+  end
+
+  def test_failed_purchase
+    @gateway.expects(:ssl_post).returns(failed_purchase_response)
+
+    response = @gateway.purchase(@amount, @credit_card, @options)
+    # puts response.params
+    assert_failure response
+    # puts response.message
+    # assert_equal Gateway::STANDARD_ERROR_CODE[:card_declined], response.error_code
+  end
+
+  def test_successful_authorize
+    @gateway.expects(:ssl_post).returns(successful_authorize_response)
+
+    response = @gateway.authorize(@amount, @credit_card, @options)
+
+    assert_success response
+    assert_equal 'ch_vfndMRFdEUac0SnBNAAT6g', response.authorization
+  end
+
+  def test_failed_authorize
+    @gateway.expects(:ssl_post).returns(failed_authorize_response)
+    response = @gateway.authorize(@amount, @credit_card, @options)
+
+    assert_failure response
+    assert_not_equal 'Approved', response.message
+  end
+
+  def test_successful_capture
+    @gateway.expects(:ssl_post).returns(successful_authorize_response)
+    response = @gateway.authorize(@amount, @credit_card, @options)
+
+    assert_success response
+
+    @gateway.expects(:ssl_post).returns(successful_capture_response)
+    response = @gateway.capture(@amount, response.authorization)
+    assert_success response
+  end
+
+  def test_failed_capture
+    @gateway.expects(:ssl_post).returns(failed_capture_response)
+    response = @gateway.capture(@amount, '')
+
+    assert_failure response
+  end
+
+  # def test_successful_refund; end
+
+  # def test_failed_refund; end
+
+  def test_successful_void
+    @gateway.expects(:ssl_post).returns(successful_authorize_response)
+    response = @gateway.authorize(@amount, @credit_card, @options)
+
+    assert_success response
+
+    @gateway.expects(:ssl_post).returns(successful_void_response)
+    response = @gateway.void(@amount, response.authorization)
+    assert_success response
+  end
+
+  def test_failed_void
+    @gateway.expects(:ssl_post).returns(failed_void_response)
+    response = @gateway.void(@amount, '')
+
+    assert_failure response
+  end
+
+  # def test_successful_verify; end
+
+  # def test_successful_verify_with_failed_void; end
+
+  # def test_failed_verify; end
+
+  # def test_scrub
+  #   assert @gateway.supports_scrubbing?
+  #   assert_equal @gateway.scrub(pre_scrubbed), post_scrubbed
+  # end
+
+  private
+
+  def pre_scrubbed
+    '
+      Run the remote tests for this gateway, and then put the contents of transcript.log here.
+    '
+  end
+
+  def post_scrubbed
+    '
+      Put the scrubbed contents of transcript.log here after implementing your scrubbing function.
+      Things to scrub:
+        - Credit card number
+        - CVV
+        - Sensitive authentication details
```

Comment:
Please add the pre and post scrubbed transcript to verify this gateway does in fact scrub appropriately